openmetadata-ui/src/main/resources/ui/playwright/utils/entity.ts

```diff
@@ -441,20 +441,43 @@ export const assignTier = async (
   endpoint: string,
   initiatorId = 'edit-tier'
 ) => {
-  await page.getByTestId(initiatorId).click();
+  // Wait for the edit button to be visible and clickable
+  const editButton = page.getByTestId(initiatorId);
+  await editButton.waitFor({ state: 'visible'});
+  await editButton.click();
+
+  // Wait for all loaders to disappear
   await page.waitForSelector('[data-testid="loader"]', { state: 'detached' });
-  const patchRequest = page.waitForResponse(`/api/v1/${endpoint}/*`);
-  await page.getByTestId(`radio-btn-${tier}`).click();
+
+  // Wait for the tier selection radio buttons to be visible
+  const tierRadioButton = page.getByTestId(`radio-btn-${tier}`);
+  await tierRadioButton.waitFor({ state: 'visible' });
+
+  // Set up response wait before clicking
+  const patchRequest = page.waitForResponse(
+    (response) =>
+      response.url().includes(`/api/v1/${endpoint}`) &&
+      response.request().method() === 'PATCH' &&
+      response.status() < 400
```

Comment:
status code should be validated at the time of `await patchRequest`

Reply:
and this is not right way to validate. lets validate actual code which we are expecting, please check all the place.